src/platform/plugins/shared/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -401,6 +403,44 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
     [dispatch, setFieldListUiState]
   );
 
+  const isSidebarHidden = useAppStateSelector((state) => {
+    return state.hideSidebar;
+  });
+
+  const toggleSidebar = useCallback(
+    (state: SidebarToggleState) => {
+      const shouldToggle = (isSidebarHidden ?? false) !== state.isCollapsed;
+      if (shouldToggle) {
+        state.toggle?.(!state.isCollapsed, false);
+      }
+    },
+    [isSidebarHidden]
+  );
+
+  useEffect(() => {
+    // After the component mounts, we want to ensure the sidebar state is correctly set for the current context
+    const currentState = sidebarToggleState$.getValue();
+    toggleSidebar(currentState);
+  }, [sidebarToggleState$, isSidebarHidden, toggleSidebar]);
+
+  useMount(() => {
+    // When this component mounts, sidebarToggleState$.toggle may not have been initialized yet.
+    // This ensures `toggle` is called and the sidebar hidden status from the context is initialized
+    const subscription = sidebarToggleState$
+      .pipe(
+        filter((state) => !!state.toggle),
+        // call toggleSidebar only once after `toggle` is initialized
+        take(1)
+      )
+      .subscribe((currentState) => {
+        toggleSidebar(currentState);
+      });
+
+    return () => {
+      return subscription.unsubscribe();
+    };
+  });
```

Comment:
I did not notice issues to the existing behaviour but looks like for metrics it does not restore the state when switching tabs yet.

![Aug-20-2025 16-20-22](https://github.com/user-attachments/assets/9ee362f4-cdd2-478f-a538-48ade2fd3003)

For the context:
- sidebar can be toggled and its last state is saved in local storage under `discover:sidebarClosed`
- when switching tabs, it should restore the last state for that tab based on `fieldListUiState`
- to enable tabs, add `feature_flags.overrides.discover.tabsEnabled` to the kibana yml

Reply:
@jughosta Correct me if I'm wrong, but it seems that to persist the sidebar state, we'll have to modify the saved object to include the `hideSidebar` flag https://github.com/crespocarlos/kibana/blob/main/src/platform/plugins/shared/saved_search/server/saved_objects/schema.ts. Should we leave this for a follow-up PR?

Reply:
I've removed the code related to the sidebar hidden status from this PR.